examples/basic/run-and-stream.ts

```diff
@@ -0,0 +1,48 @@
+import { MarketOrderSpec, GolemNetwork } from "@golem-sdk/golem-js";
+import { pinoPrettyLogger } from "@golem-sdk/pino-logger";
+
+/**
+ * Example demonstrating the execution of a command on a provider which may take a long time
+ * and returns the results from the execution during the command as a stream
+ */
+
+const order: MarketOrderSpec = {
+  demand: {
+    workload: { imageTag: "golem/alpine:latest" },
+  },
+  market: {
+    maxAgreements: 1,
+    rentHours: 0.5,
+    pricing: {
+      model: "linear",
+      maxStartPrice: 0.5,
+      maxCpuPerHourPrice: 1.0,
+      maxEnvPerHourPrice: 0.5,
+    },
+  },
+};
+
+(async () => {
+  const glm = new GolemNetwork({
+    logger: pinoPrettyLogger({
+      level: "info",
+    }),
+  });
+
+  try {
+    await glm.connect();
+    const lease = await glm.oneOf(order);
+    const exe = await lease.getExeUnit();
+
+    const remoteProcess = await exe.runAndStream("sleep 1 && echo 'Hello World' && echo 'Hello Golem' >&2");
```

Comment:
IMO it would be cool if we showcased a command that actually takes some time to complete here so the user can see the messages as they come, maybe even something as simple as:
```ts
    const remoteProcess = await exe.runAndStream(
      `
      sleep 1
      echo -n 'Hello from stdout' >&1
      echo -n 'Hello from stderr' >&2
      sleep 1
      echo -n 'Hello from stdout again' >&1
      echo -n 'Hello from stderr again' >&2
      sleep 1
      echo -n 'Hello from stdout yet again' >&1
      echo -n 'Hello from stderr yet again' >&2
      `,
    );
```

Reply:
you're right, it's better